src/wms/src/main/java/org/geoserver/wms/legendgraphic/LegendMerger.java

```diff
@@ -244,7 +244,7 @@ public static BufferedImage mergeRasterLegends(MergeOptions mergeOptions) {
         }
 
         if (layout == LegendLayout.VERTICAL) {
-            Column[] columns = createColumns(nodes, mergeOptions.getColumnHeight(), mergeOptions.getColumns());
+            Column[] columns = createColumns(nodes, mergeOptions.getColumnHeight(), mergeOptions.getColumns(), null, false);

```

Comment:
Do you know when this gets called, and why it doesn't need to truncate the legend?

Reply:
It would only apply to the one legend type